guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,605 @@
+authenticated.
+
+```handlebars
+{{!-- add to app/templates/application.hbs --}}
+
+    <button onclick={{action "logout"}}>Log Out</button>
+```
+
+Next, add the `logout` action to your `application` controller.
+
+```js
+// add to app/controllers/application.js
+
+  actions: {
+    logout() {
+      this.get('session').invalidate();
+    }
+  }
+```
+
+Clicking the "Log Out" button will take you back to the login state.
+
+### Obtaining an Access Token
+
+The final step in the process is to exchange your authorization code for an API access token. This can only be done
+securely through a back end service because it needs to know your client secret. Right now, our app looks like
+it's authenticated, but it is not _authorized_ to use the GitHub APIs. Let's add `ember-data-github` and make our
+index page show some data about the currently logged in user.
+
+```
+ember install ember-data-github
+```
+
+If you still have your `ember serve` running, you will probably want to stop and restart it to make sure the new models
+are loaded from the addon.
+
+Next, we'll add a `model` hook to our `index` route to load the user data.
+
+```js
+// add to app/routes/index.js
+
+  model() {
+    return this.get('store').findRecord('github-user', '#');
+  }
+```
+
+The `ember-data-github` addon uses "#" as a special ID value to indicate the current user.
+
+While "Me" may have been truthful in the description of the current user, it doesn't require an API call to GitHub and
+isn't very informative. Change the contents of your `index` template to show some model attributes.
+
+```handlebars
+{{!-- app/templates/index.hbs --}}
+
+<img src={{model.avatarUrl}} /> <br />
+Login: {{model.login}} <br />
+Full Name: {{model.name}}
+```
+
+If you log in to GitHub through the app now with the console open, you will see that you are not fully authorized. The
+page will be blank (we didn't do any realy error handling) and the console will tell you you received a
+`401 (Unauthorized)` for an API call to `https://api.github.com/user`. This particular API uses the identity associated
+with the authorized user to return the profile for that user, which it can't do if you're not authorized. Now that we
+have evidence that we're not truly authorized, let's fix it.
+
+We need to have a token exchange service, a service that takes an authorization code and returns an access token. You
+can implement this in any language and stack you want for writing APIs. For the demo app in development, let's implement
+this service as an Ember `http-mock`. Because it's a `POST` request, we need to add the `body-parser` package to parse
+the request body.
+
+```
+ember g http-mock token
+npm install --save-dev body-parser
+```
+
+To keep our client secret secret, we'll create a `settings.js` file in the `/server` directory looking like
+
+```js
+module.exports = {
+  CLIENT_ID: '<YOUR CLIENT ID>',
+  CLIENT_SECRET: '<YOUR CLIENT SECRET>',
+  USER_AGENT: '<YOUR APPLICATION NAME>'
+};
```

Comment:
Hm, I see what you mean but you would actually want to keep these settings in env vars also when running the server in production (think of when you're running it in a container) so the general concept of env vars makes a lot of sense for the server part in particular. Maybe it would work to use 2 separate env vars?